mmv1/third_party/terraform/transport/config.go.erb

```diff
@@ -1194,33 +1208,51 @@ func (c *Config) GetCredentials(clientScopes []string, initialCredentialsOnly bo
 		}
 	}
 
+	var creds *googleoauth.Credentials
+	var err error
 	if c.ImpersonateServiceAccount != "" && !initialCredentialsOnly {
 		opts := option.ImpersonateCredentials(c.ImpersonateServiceAccount, c.ImpersonateServiceAccountDelegates...)
-		creds, err := transport.Creds(context.TODO(), opts, option.WithScopes(clientScopes...))
+		creds, err = transport.Creds(context.TODO(), opts, option.WithScopes(clientScopes...))
 		if err != nil {
 			return googleoauth.Credentials{}, err
 		}
+	} else {
+		log.Printf("[INFO] Authenticating using DefaultClient...")
+		log.Printf("[INFO]   -- Scopes: %s", clientScopes)
 
-		return *creds, nil
+		if c.UniverseDomain != "" && c.UniverseDomain != "googleapis.com" {
+			log.Printf("[INFO]   -- Sending JwtWithScope option")
+			creds, err = transport.Creds(context.Background(), option.WithScopes(clientScopes...), internaloption.EnableJwtWithScope())
+			if err != nil {
+				return googleoauth.Credentials{}, fmt.Errorf("Attempted to load application default credentials since neither `credentials` nor `access_token` was set in the provider block.  No credentials loaded. To use your gcloud credentials, run 'gcloud auth application-default login'.  Original error: %w", err)
+			}
+		} else {
+			creds, err = transport.Creds(context.Background(), option.WithScopes(clientScopes...))
+			if err != nil {
+				return googleoauth.Credentials{}, fmt.Errorf("Attempted to load application default credentials since neither `credentials` nor `access_token` was set in the provider block.  No credentials loaded. To use your gcloud credentials, run 'gcloud auth application-default login'.  Original error: %w", err)
+			}
+		}
 	}
 
-	log.Printf("[INFO] Authenticating using DefaultClient...")
-	log.Printf("[INFO]   -- Scopes: %s", clientScopes)
-
-	if c.UniverseDomain != "" && c.UniverseDomain != "googleapis.com" {
-		log.Printf("[INFO]   -- Sending JwtWithScope option")
-		creds, err := transport.Creds(context.Background(), option.WithScopes(clientScopes...), internaloption.EnableJwtWithScope())
+	if creds.JSON != nil {
+		var content map[string]any
+		if err := json.Unmarshal([]byte(creds.JSON), &content); err != nil {
+			log.Printf("[WARN] error unmarshaling credentials, skipping Universe Domain detection")
```

Comment:
This will preserve the c.UniverseDomain value. Could that result in an issue?

Reply:
c.UniverseDomain would default to the configured value in this case. In this unexpected case where we could not verify, would we want to lean towards going with the configured value in case it is actually valid?

Reply:
I would think allowing false positives (the universe domain does not match) would be less disruptive for users than blocking false negatives (the universe domains match but we could not unmarshal the JSON)

Reply:
> would we want to lean towards going with the configured value in case it is actually valid?

No- sending a request to the wrong universe_domain exposes credentials to an unknown server, which is what we need to guard against. This probably has to always fail (i.e. create a false positive), although I'm not sure how we verify it would be backwards-compatible to do so.

Reply:
Understood, I will unset the c.UniverseDomain so it can fall through and eventually fail if there is a mismatch.

As far as backwards-compatibility, right now any case that is not a credentials file already leads to the mismatch error due to the bug from the previous PR. So at worse, we would carry forward the same error in this hopefully rare case.